nbextensions/upload_it/main.js

```diff
@@ -79,14 +79,26 @@ define([
             window.console.log("Uploading ", notebook.notebook_path, " to ", url, formdata);
             $.ajax({
               url: url,
+              xhrFields: {withCredentials: true},
               data: formdata,
               contentType: false,
               processData: false,
               method: "POST",
               success: function(data, status, jqXHR) {
-                window.console.log("Upload OK", data);
+                // Open the report in a new tab.
+                var u = new URL(url);
+                var reportURL = u.protocol + "//" + u.host + data;
+                window.open(reportURL, '_blank');
+                window.console.log("Upload OK", reportURL);
               },
               error: function(jqXHR, status, err) {
+                if (err == "Unauthorized") {
+                  window.console.log("Unauthorized, attempting login");
+                  let loginURL = new URL(configuration.upload_it_server_url);
```

Comment:
You can use `const` for loginURL as well.

Reply:
No, it is modified in the line 'loginURL.pathname = '/login';'

Reply:
Not really. `const` just prevents reassignment of variables. [MDN doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#Description) says 
> the object's contents (e.g., its properties) can be altered.

```javascript
const x = {};
x.foo = "foo"; // OK
x = {}         // Error
```